crypto/fipsmodule/sshkdf/kdf.c

```diff
@@ -0,0 +1,112 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+// Based on OpenSSL 3.x SSHKDF().
```

Comment:
If the code was mostly taken from OpenSSL, we should put their licenses in the corresponding files. We should be able to use the license header they have here:
https://github.com/openssl/openssl/blob/ad4910fad22d57c6d685b0ca83bdb7b2bf69d8fd/providers/implementations/kdfs/sshkdf.c#L1-L8
It might also help to specify which test cases were taken from OpenSSL as well.

Reply:
Good idea, I wasn't sure what to do in this case.